src/test/test-async.c

```diff
@@ -17,21 +19,44 @@ static void *async_func(void *arg) {
 }
 
 TEST(test_async) {
+        assert_se(asynchronous_job(async_func, NULL) >= 0);
+        assert_se(asynchronous_sync(NULL) >= 0);
+
+        assert_se(test_async);
+}
+
+TEST(asynchronous_close) {
         _cleanup_(unlink_tempfilep) char name[] = "/tmp/test-asynchronous_close.XXXXXX";
-        int fd;
+        int fd, r;
 
         fd = mkostemp_safe(name);
         assert_se(fd >= 0);
         asynchronous_close(fd);
 
-        assert_se(asynchronous_job(async_func, NULL) >= 0);
-        assert_se(asynchronous_sync(NULL) >= 0);
-
         sleep(1);
 
         assert_se(fcntl(fd, F_GETFD) == -1);
         assert_se(errno == EBADF);
-        assert_se(test_async);
+
+        r = safe_fork("(subreaper)", FORK_RESET_SIGNALS|FORK_CLOSE_ALL_FDS|FORK_DEATHSIG|FORK_LOG|FORK_WAIT, NULL);
+        assert(r >= 0);
+
+        if (r == 0) {
+                /* child */
+
+                assert(prctl(PR_SET_CHILD_SUBREAPER, 1, 0, 0, 0) >= 0);
+
+                fd = open("/dev/null", O_RDONLY|O_CLOEXEC);
+                assert_se(fd >= 0);
+                asynchronous_close(fd);
+
+                sleep(1);
```

Comment:
Normally 1 s would be enough, but this could be an issue in overloaded CI. But let's see.

Reply:
this is pre-existing. I don't have any nice idea how to do this better...